demisto_sdk/__main__.py

```diff
@@ -1439,26 +1448,63 @@ def update_release_notes(**kwargs):
               is_flag=True)
 @click.option("--use-pack-metadata", help="Whether to update the dependencies from the pack metadata.", required=False,
               is_flag=True)
+@click.option("--all-packs-dependencies", help="Return a json file with ALL content packs dependencies. "
+                                               "The json file will be saved under the path given in the "
+                                               "'--output-path' argument", required=False, is_flag=True)
+@click.option("-o", "--output-path", help="The destination path for the packs dependencies json file. This argument is "
+                                    "only relevant for when using the '--all-packs-dependecies' flag.", required=False)
+@click.option("--get-dependent-on", help="Get only the packs dependent ON the given pack. Note: this flag can not be"
+                                         " used for the packs ApiModules and Base", required=False,
+              is_flag=True)
 def find_dependencies(**kwargs):
     """Find pack dependencies and update pack metadata."""
     check_configuration_file('find-dependencies', kwargs)
     update_pack_metadata = not kwargs.get('no_update')
-    input_path: Path = Path(kwargs["input"])  # To not shadow python builtin `input`
+    input_path = kwargs.get('input')
     verbose = kwargs.get('verbose', False)
     id_set_path = kwargs.get('id_set_path', '')
     use_pack_metadata = kwargs.get('use_pack_metadata', False)
-    if len(input_path.parts) != 2 or input_path.parts[-2] != "Packs":
-        print_error(f"Input path ({input_path}) must be formatted as 'Packs/<some pack name>'. "
-                    f"For example, Packs/HelloWorld")
-        sys.exit(1)
+    all_packs_dependencies = kwargs.get('all_packs_dependencies')
+    get_dependent_on = kwargs.get('get_dependent_on')
+    output_path = kwargs.get('output_path', ALL_PACKS_DEPENDENCIES_DEFAULT_PATH)
+
     try:
-        PackDependencies.find_dependencies(
-            pack_name=input_path.name,
-            id_set_path=str(id_set_path),
-            verbose=verbose,
-            update_pack_metadata=update_pack_metadata,
-            use_pack_metadata=use_pack_metadata
-        )
+        if all_packs_dependencies:
```

Comment:
avoid adding logic to this module, it is harder to test.
move it to the command module.

Reply:
Done

Reply:
Done, added a manager